gobblin-core/src/main/java/gobblin/converter/string/StringFilterConverter.java

```diff
@@ -0,0 +1,65 @@
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.converter.string;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Strings;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.WorkUnitState;
+import gobblin.converter.Converter;
+import gobblin.converter.DataConversionException;
+import gobblin.converter.EmptyIterable;
+import gobblin.converter.SchemaConversionException;
+import gobblin.converter.SingleRecordIterable;
+
+
+/**
+ * Implementation of {@link Converter} which filters strings based on whether or not they match a regex specified by
+ * {@link ConfigurationKeys#CONVERTER_FILTER_STRINGS_BY}
+ */
+public class StringFilterConverter extends Converter<Class<String>, Class<String>, String, String> {
+
+  private Pattern pattern;
+  private Optional<Matcher> matcher;
+
+  @Override
+  public Converter<Class<String>, Class<String>, String, String> init(WorkUnitState workUnit) {
+    this.pattern =
+        Pattern.compile(Strings.nullToEmpty(workUnit.getProp(ConfigurationKeys.CONVERTER_FILTER_STRINGS_BY)));
+    this.matcher = Optional.absent();
```

Comment:
It's not strictly necessary since the scope of the variable is only this class. It doesn't look like you can initialize a blank matcher object in Java, it has to be tied with some input string. That input string is not known until the convertRecord() method gets executed. So the first time it is executed the matcher object is initialized with the first input record seen. Every time after than it calls matcher.reset(inputRecord).